connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectAddArtifactsHandler.scala

```diff
@@ -0,0 +1,246 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.connect.service
+
+import com.google.common.io.CountingOutputStream
+import io.grpc.stub.StreamObserver
+import java.nio.file.{Files, Path, Paths}
+import java.util.zip.{CheckedOutputStream, CRC32}
+import scala.collection.mutable
+import scala.util.control.NonFatal
+
+import org.apache.spark.connect.proto
+import org.apache.spark.connect.proto.{AddArtifactsRequest, AddArtifactsResponse}
+import org.apache.spark.connect.proto.AddArtifactsResponse.ArtifactSummary
+import org.apache.spark.sql.connect.artifact.SparkConnectArtifactManager
+import org.apache.spark.util.Utils
+
+/**
+ * Handles [[AddArtifactsRequest]]s for the [[SparkConnectService]].
+ *
+ * @param responseObserver
+ */
+class SparkConnectAddArtifactsHandler(val responseObserver: StreamObserver[AddArtifactsResponse])
+    extends StreamObserver[AddArtifactsRequest] {
+
+  // Temporary directory where artifacts are rebuilt from the bytes sent over the wire.
+  protected val stagingDir: Path = Utils.createTempDir().toPath
```

Comment:
Why are these protected?

Reply:
Makes it easier for testing to inspect the state (used in `TestAddArtifactsHandler` [here](https://github.com/apache/spark/blob/4bb72960465e863986c10ba381e8d9ac27009151/connector/connect/server/src/test/scala/org/apache/spark/sql/connect/service/AddArtifactsHandlerSuite.scala#L46))